cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -501,16 +505,41 @@ func main() {
 	}
 	logging.Infof("Ready for new connections")
 
-	(&proxy.Client{
+	proxyClient := &proxy.Client{
 		Port:           port,
 		MaxConnections: *maxConnections,
 		Certs: certs.NewCertSourceOpts(client, certs.RemoteOpts{
-			APIBasePath:  *host,
-			IgnoreRegion: !*checkRegion,
-			UserAgent:    userAgentFromVersionString(),
-			IPAddrTypeOpts:   ipAddrTypeOptsInput,
+			APIBasePath:    *host,
+			IgnoreRegion:   !*checkRegion,
+			UserAgent:      userAgentFromVersionString(),
+			IPAddrTypeOpts: ipAddrTypeOptsInput,
 		}),
 		Conns:              connset,
 		RefreshCfgThrottle: refreshCfgThrottle,
-	}).Run(connSrc)
+	}
+
+	if *usegRPC {
```

Comment:
this branch is pretty unbalanced. Could you either push this into proxyClient.Run or maybe add a proxyClent.RunGrpc(connSrc) instead?

Reply:
We talked about this in person a little, the broader point here is that this proxyClient and the grpcproxy should have a more parallel structure. One way to achieve that may be to extract the config retrieving part.